src/cli/parser.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * Command line argument parser for srvx CLI
+ */
+
+export interface CLIOptions {
+  /** The file to serve */
+  file: string;
+  /** Server port */
+  port?: number;
+  /** Server hostname */
+  host?: string;
+  /** Enable HTTPS */
+  https?: boolean;
+  /** TLS certificate file path */
+  cert?: string;
+  /** TLS private key file path */
+  key?: string;
+  /** Enable development mode (watch mode in future) */
+  dev?: boolean;
+  /** Show help */
+  help?: boolean;
+  /** Show version */
+  version?: boolean;
+  /** Runtime to use (auto-detected if not specified) */
+  runtime?: "node" | "deno" | "bun";
+}
+
+/**
+ * Parse command line arguments
+ */
+export function parseArgs(args: string[]): CLIOptions {
```

Comment:
As both Deno and Bun support Node compat, we can use Node.js [util.parseArgs](https://nodejs.org/api/util.html#utilparseargsconfig)